frontend/_data/segments.js

```diff
@@ -98,10 +100,52 @@ const allSegmentData = async () => {
       }
     })
   );
+
+  await addObservations(segmentData);
+
   saveCache();
   return augmentedSegmentData;
 };
 
+const addObservations = async (segmentData) => {
```

Comment:
suggestion: Maybe this would be better named `addWeatherObservations` or something similar? `observations` is a little too generic IMO especially when we're observing Strava counts elsewhere in the code